### DISCUSSION TITLE
v0.22.0

### DISCUSSION THREAD

**Discussion by erogol** [coqui-ai/TTS#3413] [Versions & Releases]
## What's Changed
* fix: Few typos in Tortoise docs. by @VladCuciureanu in https://github.com/coqui-ai/TTS/pull/3352
* fix pause problem of Chinese speech by @aaron-lii in https://github.com/coqui-ai/TTS/pull/3351
* Fix typos by @omahs in https://github.com/coqui-ai/TTS/pull/3368
* Print message for either commercial license or CPML by @JRMeyer in https://github.com/coqui-ai/TTS/pull/3381
* Add inference parameters by @WeberJulian in https://github.com/coqui-ai/TTS/pull/3373
* Training fastspeech2 with External Speaker Embeddings by @freds0 in https://github.com/coqui-ai/TTS/pull/3404
* fixes a typo by @joelhoward0 in https://github.com/coqui-ai/TTS/pull/3392
* support multiple GPU training for XTTS by @aaron-lii in https://github.com/coqui-ai/TTS/pull/3391
* Add studio speakers to open source XTTS! by @WeberJulian in https://github.com/coqui-ai/TTS/pull/3405

## New Contributors
* @VladCuciureanu made their first contribution in https://github.com/coqui-ai/TTS/pull/3352
* @aaron-lii made their first contribution in https://github.com/coqui-ai/TTS/pull/3351
* @omahs made their first contribution in https://github.com/coqui-ai/TTS/pull/3368
* @JRMeyer made their first contribution in https://github.com/coqui-ai/TTS/pull/3381
* @freds0 made their first contribution in https://github.com/coqui-ai/TTS/pull/3404
* @joelhoward0 made their first contribution in https://github.com/coqui-ai/TTS/pull/3392

**Full Changelog**: https://github.com/coqui-ai/TTS/compare/v0.21.3...v0.22.0

<hr /><em>This discussion was created from the release <a href='https://github.com/coqui-ai/TTS/releases/tag/v0.22.0'>v0.22.0</a>.</em>

**Comment by erew123** [comment]
I have a question, not absolutely specific to this version, more general. I am doing something with text splitting and I am trying to avoid giving a very long question, so here is the simple question. 

Are there any special characters that in the standard ASCII English character set, that are **not punctuation**, other than the asterisk *****, that the TTS engine **will disregard**/throw away and **NOT** try to generate speech? 

I know the asterisk is fine, but I need at least one more special character that I can use for my text splitting reasons! Sorry if there is documentation on this, but I couldn't find it.

Thanks

EDIT - I actually managed to figure out quite a clever method (well, that's my opinion at least) for manipulating AI model textual outputs to be able to split them out and do what I needed to do. Whilst I found a way around the issue. I can say it would have been nice to have a a couple of special characters of some kind in the higher end of the ASCII character set that the TTS text parsing engine would just throw away/disregard as this would make splitting narrator/character/other voices in 1x generation much easier, when an AI model throws out text (very complicated to explain without a long message on here). In short, its to do with text spitting and being able to do that, before you ship things to the TTS engine.

**Comment by unifirer** [comment]
> I have a question, not absolutely specific to this version, more general. I am doing something with text splitting and I am trying to avoid giving a very long question, so here is the simple question.
> 
> Are there any special characters that in the standard ASCII English character set, that are **not punctuation**, other than the asterisk *****, that the TTS engine **will disregard**/throw away and **NOT** try to generate speech?
> 
> I know the asterisk is fine, but I need at least one more special character that I can use for my text splitting reasons! Sorry if there is documentation on this, but I couldn't find it.
> 
> Thanks
> 
> EDIT - I actually managed to figure out quite a clever method (well, that's my opinion at least) for manipulating AI model textual outputs to be able to split them out and do what I needed to do. Whilst I found a way around the issue. I can say it would have been nice to have a a couple of special characters of some kind in the higher end of the ASCII character set that the TTS text parsing engine would just throw away/disregard as this would make splitting narrator/character/other voices in 1x generation much easier, when an AI model throws out text (very complicated to explain without a long message on here). In short, its to do with text spitting and being able to do that, before you ship things to the TTS engine.

bro im a big fan of ur new extension, havent used it but it sounds way better, i just found u on reddit 5 mins and i got out of bed to install it. coincidentally found u here while trying to figure out how to check model.pth. version.

i read ur edit thrice and i couldnt comprehend(ure truly technically minded) but im not sure if [ or ( will help? also i discovered the opening qoute " and closing one and the universal one, there are 3 versions. in unicode 8.  the opening one and closing one gave me issues for speaking

**Comment by erew123** [comment]
@unifirer Ive just added one small update just o capture another possible outlier situation with how the text is processed. As for the problem, Ive found a very unconventional, but working and flexible, way around it.

The issue was a tough mix between how AI models send out text (asterisks, double quote and nothing around sentences), how you differentiate those between narrator/character esp the bits with no asterisks or double quotes, then the python split command, which strips quotes off, leaving you with only the asterisk surrounded lines and the other 2x kinds now look the same, and then, if you pass any "special" characters over to the TTS engine, thats what appears to make many of the strange sounds, so you need to strip them off too, before you send it to the TTS engine (which is why mine is sounding better as Ive stripped all special characters (including quotes and asterisks) off the text before it hits the TTS engine).......and thats the short story. So somehow, you need to strip everything off and still let is know who is who....So, having some characters that the text engine will just ignore 100% and not make strange sounds, would have been perfect. but Ive got a solution at least. 

Thanks for the suggestion though :) I did give a load of other characters a go, but I ended up back at the strange sounds occurring, hence coming up with this other method.

You may want to update the script.py file again, though its not a biggy!

Post back on my github if youve got any problems with it and I hope it works well for you :)
